src/java.base/share/classes/java/io/RandomAccessFile.java

```diff
@@ -81,6 +89,30 @@ public class RandomAccessFile implements DataOutput, DataInput, Closeable {
     private static final int O_DSYNC =  8;
     private static final int O_TEMPORARY =  16;
 
+    /**
+     * Calculate the size of a temporary direct buffer.
+     *
+     * @param the number of bytes in question
+     * @return the size of buffer to use
+     */
+    // package scope
+    static final int bufferSize(int len) {
+        if (len <= 0) {
+            return 0;
+        }
+
+        if (len >= MAX_BUFFER_SIZE) {
+            return MAX_BUFFER_SIZE;
+        }
+
+        if (len % MIN_BUFFER_SIZE == 0) {
+            return len;
+        }
+
+        int multiple = (len + MIN_BUFFER_SIZE - 1)/MIN_BUFFER_SIZE;
```

Comment:
isn't that equivalent `(len / MIN_BUFFER_SIZE) + 1`

Reply:
Yes, I think you are correct. Good catch!

Reply:
So changed in 9d7e4fbf39668ff3938b5f0f35d2e997ebbd9777.